document system features (#8706)
NixOS/nix: doc/manual/src/language/advanced-attributes.md

    [`outputHashAlgo`](#adv-attr-outputHashAlgo)
    like for *fixed-output derivations* (see above).

>     It also implicitly requires that the machine to build the derivation must have the `ca-derivations` [system feature](@docroot@/command-ref/conf-file.md#conf-system-features).

Reviewer: I think this is the proper grammatical form:

```suggestion
    It also implicitly requires that the machine to build the derivation on have the `ca-derivations` [system feature](@docroot@/command-ref/conf-file.md#conf-system-features) enabled.
```

    This is useful, for example, when generating self-contained filesystem images with
    their own embedded Nix store: hashes found inside such an image refer
    to the embedded store and not to the host's Nix store.

- [`requiredSystemFeatures`]{#adv-attr-requiredSystemFeatures}\

>   If a derivation has the `requiredSystemFeatures` attribute, then Nix will only build it on a machine that has the corresponding features set in its [`system-features` configuration](@docroot@/command-ref/conf-file.md#conf-system-features).

Reviewer 2: These have to be repeated in the remote builder configuration as well; for remote builds it won't look at the remote's configuration, [which is unfortunate but also simpler and low latency.]

Author: This is stuff for the remote builds reference page. That needs to be reworked anyway.

Reviewer 2: Doesn't mean that we can present false information here.

I suppose you could say
```suggestion
  If a derivation has the `requiredSystemFeatures` attribute, then Nix will only build it locally if the [`system-features` configuration](@docroot@/command-ref/conf-file.md#conf-system-features) contains the corresponding features.
```

although that's still partial information to a degree that it's confusing. Best to just mention remote builds here.

  For example, setting

  ```nix
  requiredSystemFeatures = [ "kvm" ];
  ```

>   ensures that the derivation can only be built on a machine with the `kvm` feature.

Reviewer 2: This is correct even for remote builds. An error would be raised if misconfigured.

    This is useful, for example, when generating self-contained filesystem images with
    their own embedded Nix store: hashes found inside such an image refer
    to the embedded store and not to the host's Nix store.

- [`requiredSystemFeatures`]{#adv-attr-requiredSystemFeatures}\

>   If a derivation has the `requiredSystemFeatures` attribute, then Nix will only build it on a machine that has the corresponding features set in its [`system-features` configuration](@docroot@/command-ref/conf-file.md#conf-system-features).

Reviewer: I think one enables a feature rather than setting it. 

```suggestion
  If a derivation has the `requiredSystemFeatures` attribute, then Nix will only build it on a machine that has the corresponding features enabled in its [`system-features` configuration](@docroot@/command-ref/conf-file.md#conf-system-features).
```

Author: "System features" are not something that can be turned on, they are just strings to match between a derivation and a build machine. Their meaning is defined by the user. The special cases listed here are exceptions.

  For example, setting

  ```nix
  requiredSystemFeatures = [ "kvm" ];
  ```

>   ensures that the derivation can only be built on a machine with the `kvm` feature.

Reviewer: ```suggestion
  ensures that the derivation can only be built on a machine with the `kvm` feature enabled.
```

Reviewer 3: I think the current state fits better since you can't enable kvm on every machine (for example on some vps's).
